test/test_macos_archs.py

```diff
@@ -0,0 +1,69 @@
+import platform
+
+import pytest
+
+from . import test_projects, utils
+
+basic_project = test_projects.new_c_project()
+
+ALL_MACOS_WHEELS = (
+    utils.expected_wheels('spam', '0.1.0', machine_arch='x86_64')
+    + utils.expected_wheels('spam', '0.1.0', machine_arch='arm64')
+)
+
+
```

Comment:
By the way, I've been able to run these tests locally on my MacBook Air, and they pass on arm64 too. But I don't think the next release of cibuildwheel will officially include arm64 host support, because we can't run the entire test suite on arm64 yet. We might need to do a few tweaks once arm64 CI is available before declaring full support. The arm64 parts of this PR are mostly to nail down a design and show the intention.